pkg/apis/cr/v1alpha1/types.go

```diff
@@ -307,3 +308,93 @@ type ProfileList struct {
+type RepositoryServerSpec struct {
+	Storage    Storage    `json:"storage,omitempty"`
+	Repository Repository `json:"repository,omitempty"`
+	Server     Server     `json:"server,omitempty"`
+}
+
+// Storage references the backend store where a repository already exists
+// and the credential necessary to connect to the backend store
+type Storage struct {
+	Secret           corev1.SecretReference `json:"secret,omitempty"`
+	CredentialSecret corev1.SecretReference `json:"credentialSecret,omitempty"`
+}
+
+// Repository details for the purpose of establishing a connection
+type Repository struct {
+	RootPath       string                 `json:"rootPath,omitempty"`
+	PasswordSecret corev1.SecretReference `json:"passwordSecret,omitempty"`
+}
+
+// Server details required for starting the repository proxy server and initializing the repository client users
+type Server struct {
+	AdminSecret      corev1.SecretReference `json:"adminSecret,omitempty"`
+	TLSSecret        corev1.SecretReference `json:"tlsSecret,omitempty"`
+	UserAccessSecret corev1.SecretReference `json:"userAccessSecret,omitempty"`
+	LabelSelector    metav1.LabelSelector   `json:"labelSelector,omitempty"`
+}
+
+// RepositoryServerStatus is the status for the RepositoryServer. This should only be updated by the controller.
+type RepositoryServerStatus struct {
+	Conditions []Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	ServerInfo ServerInfo  `json:"serverInfo,omitempty"`
+}
+
+// Condition contains details of the current state of the RepositoryServer resource
+type Condition struct {
+	LastTransitionTime metav1.Time                   `json:"lastTransitionTime,omitempty"`
+	LastUpdateTime     metav1.Condition              `json:"lastUpdateTime,omitempty"`
+	Status             metav1.ConditionStatus        `json:"status"`
+	Type               RepositoryServerConditionType `json:"type"`
+}
+
+// RepositoryServerConditionType defines all the various condition types of the RepositoryServer resource
+type RepositoryServerConditionType string
+
+const (
+	// RepositoryReady indicates whether the existing repository is connected and ready to use
+	RepositoryReady RepositoryServerConditionType = "RepositoryReady"
+
+	// ServerInitialized means that the proxy server, that serves the repository, has been started
+	ServerInitialized RepositoryServerConditionType = "ServerInitialized"
+
+	// ClientsInitialized indicates that the client users have been added or updated to the repository server
+	ClientsInitialized RepositoryServerConditionType = "ClientsInitialized"
+
+	// ServerRefreshed denotes the refreshed condition of the repository server in order to register client users
+	ServerRefreshed RepositoryServerConditionType = "ServerRefreshed"
+
+	// ServerReady represents the ready state of the repository server and the pod which runs the proxy server
+	ServerReady RepositoryServerConditionType = "ServerReady"
```

Comment:
Should we add Failure status types as well here?
For e.g. `ServerStopped` etc?

Reply:
Yes 💯 I will add a Failure status here 👍🏼